### DISCUSSION TITLE
4k Display

### DISCUSSION THREAD

**Discussion by wolfgang-ch** [mytourbook/mytourbook#1389] [General]
I had to buy a new monitor, the background light of the old monitor flickered, now it's a 4k monitor and found several sizing issues for MT.

When MT do not recognize the 4k monitor, which happend when Windows scaling was 150%, then this parameter in the commandline can force scaling to 200% for MT, this could also be another value, just test it.

`-vmargs -Dswt.autoScale=200`

All MT icons are badly scaled but it would be a lot of work to manually create all enlarged MT icons. Each icon can have up to 4 images for light/dark theme and both can be disabled. I did it now for the tour chart icon in the next version 24.8

![image](https://github.com/user-attachments/assets/2950f080-7d7c-47db-8c19-07e91eed4178)

**Comment by wolfgang-ch** [comment]
Some of the `autoScale` parameter values are causing a resize issue on the newly expandable slideouts

* Map point
* Photo tooltip

When resizing the dialog in both directions, all is OK, when resizing in just one direction, the dialog increases constantly into the other direction

These are the results with the `autoScale` parameter:

* 150 😠 
* 160 OK 
* 170 OK
* 180 😠 
* 190 OK
* 200 OK

https://github.com/user-attachments/assets/e590f6ec-d181-4eae-8c29-8139bf1997fb

**Reply by wolfgang-ch** [reply]
Created an issue https://github.com/mytourbook/mytourbook/issues/1393

**Comment by wolfgang-ch** [comment]
> All MT icons are badly scaled but it would be a lot of work to manually create all enlarged MT icons. Each icon can have up to 4 images for light/dark theme and both can be disabled. I did it now for the tour chart icon in the next version 24.8

I found a "shortcut" to create 4k icons faster by copying an icon and setting the dpi to 192 and save it with padded `@2x` name. It's still some work but the UI looks much better.

![image](https://github.com/user-attachments/assets/647c4ee2-9920-47d9-b519-87594b090c32)

**Comment by wolfgang-ch** [comment]
Rendering a 2D map with 4k tiles, they are automatically zoomed (because of SWT) and clipped (because of default tile size 256 pixel)

![image](https://github.com/user-attachments/assets/202f8f19-2aa5-4493-a6f9-a11fb57539a3)

**Reply by wolfgang-ch** [reply]
4k tiles are now supported

![image](https://github.com/user-attachments/assets/ef20fe19-7383-4b5e-97fe-4b6eb01412f6)

**Comment by mff73** [comment]
quick off topic question : is there already a way or a post discussion about MTB being able to use 512x512 tiles ? Or any trick to use some urls which are providing 512 pixel tiles ?
Thanks in adance. If too offtopic, i will open a new discussion for this.

**Reply by wolfgang-ch** [reply]
In the above screenshot https://github.com/mytourbook/mytourbook/discussions/1389#discussioncomment-10470620 you can see, that just adding `@2x` in the tile image url will support 512x512 tile images

**Reply by mff73** [reply]
hehe, i already tried, but it doesn't work with the url tiles i am trying to use 😊 (and not sure i can share it online here)
